### PR TITLE
Prompt user to confirm exit on application quit

### LINKED ISSUE #1116: Warning dialog on GUI quit
I think a worthwhile feature would be a warning dialog asking the user if they are sure they want to quit upon catching one of the various ways of sending the terminate signal to the GUI.  I have on more than one occasion killed the GUI accidentally (by an errant ctrl+q or otherwise...).

### PR DESCRIPTION
**Pull request by bnmajor**
Fixes: #510
Fixes #1116

### CODE REVIEW

File: hexrd/ui/main_window.py [hunk 1]
```diff
@@ -747,6 +747,14 @@ def update_config_gui(self):
 
     def eventFilter(self, target, event):
         if type(target) == QMainWindow and event.type() == QEvent.Close:
+            msg = 'Are you sure you want to quit?'
+            buttons = (QMessageBox.No | QMessageBox.Yes)
+            default = QMessageBox.No
+            response = QMessageBox.question(
+                self.ui, 'HEXRD', msg, buttons, default)
```
Comment (psavery) [inline hexrd/ui/main_window.py:754]: It looks to me that the `buttons` and `default` arguments are the same as the defaults for `QMessageBox.question()`. Maybe we can remove them?

**Review by psavery — COMMENTED** [review]
It works for me in my testing! I guess we just need to figure out how to make it work with the CI.

File: hexrd/ui/main_window.py (hunk 1 above)
Comment (bnmajor) [reply hexrd/ui/main_window.py]: Yes, thank you. Removed.

**Comment by joelvbernier** [comment]
@bnmajor and @psavery -- looks like we have some CI issues to address?

**Review by psavery — APPROVED** [review]
Looks excellent to me!

**Pull request merged**